src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -69,8 +69,10 @@ def put(self, request, group, integration_id):
         installation = integration.get_installation(organization_id)
         try:
             data = installation.get_issue(external_issue_id, data=request.DATA)
+        except IntegrationFormError as exc:
+            return Response(exc.field_errors, status=400)
         except IntegrationError as exc:
-            return Response({'detail': exc.message}, status=400)
+            return Response({'non_field_errors': [exc.message]}, status=400)
```

Comment:
It'd be nice for error payloads to be consistent in some way (meaning a consistent top-level envelope). Quick, probably not thought out, example.

```
{
  errors: [{
    type: 'non_field_error',
    detail: exc.message
  }, {
    type: 'externalIssue',
    detail: 'Issue ID is required'
  }],
}
```

Reply:
cc @billyvg

Reply:
So we have this documented, but it's actually hardly used anywhere: https://docs.sentry.io/internal/api

In `src/sentry/api/exceptions.py` we have some exceptions that include `code`, `message`, and `extra`, we don't handle multiple errors though.

Reply:
Using `errors` as the top level key for multiple errors lets clients more easily handle 1/N error scenarios. Having the each item in the list of errors use the same structure as a single error would be :+1:

Reply:
Definitely agree, but we'll need to go back and cleanup various API endpoints for this

Reply:
We don't really treat the API as a contract today. It's mostly a change as needed situation. I'm planning on changing this in the future, but for now it's not really a concern, for better or worse.